public/app/core/components/Page/PageHeader.tsx

```diff
@@ -57,6 +57,7 @@ const getStyles = (theme: GrafanaTheme2) => {
     title: css({
       display: 'flex',
       flexDirection: 'row',
+      maxWidth: '100%',
```

Comment:
Added this to support custom title rendering where the wrapper is a `<Stack>` component – tested with various different usages of `<Page>` and didn't find any regressions :)